module/userProjects/helper.js

```diff
@@ -8,507 +8,658 @@
+          },
+          {
+            "role": "CONTENT_CREATOR",
+            "users": [
+              "<contentCreatorUserId>"
+            ]
+          },
+          {
+            "role": "COURSE_MENTOR",
+            "users": [
+              "<courseMentorUserId>"
+            ]
+          }
+        ],
+        "action": "delete-user",
+        "iteration": 1
+      }
+    }
+   * @returns {Promise} success Data.
+   */
+  static userDelete(userDeleteEvent) {
+    return new Promise(async (resolve, reject) => {
+      let userId = userDeleteEvent.edata.userId;
+      let userProfileUpdateData = [];
+      let userProjectData = await projectQueries.projectDocument(
+        {
+          userId: userId,
+          userProfile: { $exists: true },
+        },
+        ["userProfile"],
+        "none"
+      );
+      if (userProjectData.length === 0) {
+        return resolve({ success: false });
+      }
+      if (userProjectData.length > 0) {
+        userProjectData.forEach((userProfile) => {
+          let updateProfile = {
+            updateOne: {
+              filter: { _id: userProfile._id },
+              update: {
+                $set: {
+                  "userProfile.firstName": CONSTANTS.common.DELETED_USER,
+                },
+                $unset: {
+                  "userProfile.email": 1,
+                  "userProfile.maskedEmail": 1,
+                  "userProfile.maskedPhone": 1,
+                  "userProfile.recoveryEmail": 1,
+                  "userProfile.phone": 1,
+                  "userProfile.lastName": 1,
+                  "userProfile.prevUsedPhone": 1,
+                  "userProfile.prevUsedEmail": 1,
+                  "userProfile.recoveryPhone": 1,
+                },
+              },
+              multi: true,
+            },
+          };
+          userProfileUpdateData.push(updateProfile);
+        });
+      }
+      let updateUserProfile = await projectQueries.bulkUpdate(
+        userProfileUpdateData
+      );
+      if (updateUserProfile) {
+        /**
+         * Telemetry Raw Event
+         * {"eid":"","ets":1700188609568,"ver":"3.0","mid":"e55a91cd-7964-46bc-b756-18750787fb32","actor":{},"context":{"channel":"","pdata":{"id":"projectservice","pid":"manage-learn","ver":"7.0.0"},"env":"","cdata":[{"id":"adf3b621-619b-4195-a82d-d814eecdb21f","type":"Request"}],"rollup":{}},"object":{},"edata":{}}
+         */
+        let rawEvent = await UTILS.getTelemetryEvent();
+        rawEvent.eid = CONSTANTS.common.AUDIT;
+        rawEvent.context.channel = userDeleteEvent.context.channel;
+        rawEvent.context.env = "User";
+        rawEvent.edata.state = CONSTANTS.common.DELETE_STATE;
+        rawEvent.edata.type = CONSTANTS.common.USER_DELETE_TYPE;
+        rawEvent.edata.props = [];
+        let userObject = {
+          id: userId,
+          type: "User",
+        };
+        rawEvent.actor = userObject;
+        rawEvent.object = userObject;
+
+        let telemetryEvent = {
```

Comment:
@ankitpws Make this an another utils function

Reply:
created new function